openedx/openedx-platform
# feat: mgmt cmd to replace v1 libr refs in courses

## Description
This PR adds a management command to, given a mapping of V1 content libraries to matching v2 content libraries, replaces references to v1 libs in courses (in library source xblocks) with V2 libraries. It does so by manipulating the mongo document directly.

it also offers some improvements to the management command which copies all v1 libraries into v2 libraries.

## Supporting information
https://2u-internal.atlassian.net/browse/TNL-10638

## Testing instructions

1. Setup V1 libraries as test data.
2. Use some of those v1 libraries in at least 1 or 2 courses.
3. Then setup V2 libraries using https://github.com/openedx/frontend-app-library-authoring/tree/master#introduction
4. Then run from devstack:
```
> $ make dev.shell.studio
> $  ./manage.py cms copy_libraries_from_v1_to_v2 '11111111-2111-4111-8111-111111111111' ./mapping.csv --all
> $  ./manage.py cms replace_v1_lib_refs_with_v2_in_courses ./mapping.csv
```
5. All courses' library source blocks should now point to v2 libraries. You can inspect this by clicking edit on the block.

## Deadline

"None"
